src/vi/comanhua/src/eu/kanade/tachiyomi/extension/vi/comanhua/CoManhua.kt

```diff
@@ -0,0 +1,164 @@
+        .rateLimit(3)
+        .build()
+
+    override val popularPath = "truyen-de-cu"
+
+    override fun popularMangaRequest(page: Int): Request {
+        val offset = (page - 1) * 40
+        val url = "$baseUrl/$popularPath?offset=$offset"
+        return GET(url, headers)
+    }
+
+    override fun popularMangaSelector() = "div.pda.manga-list div.manga-item"
+
+    override fun popularMangaFromElement(element: Element) = SManga.create().apply {
+        element.select("div.manga-title a").let {
+            title = it.text()
+            setUrlWithoutDomain(it.attr("abs:href"))
+        }
+        thumbnail_url = imageOrNull(element.selectFirst("div.manga-image img")!!)
+    }
+
+    override fun popularMangaNextPageSelector() = "div.list-pagination a:last-child:not(.active)"
+
+    override val queryParam = "name"
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        val url = "$baseUrl/tim-truyen".toHttpUrl().newBuilder()
+
+        url.addQueryParameter("chapter", "")
+        url.addQueryParameter("year", "")
+        url.addQueryParameter("name", query)
+
+        var statusFilter = "Tất cả"
+        var genreFilter = ""
+
+        filters.forEach { filter ->
+            when (filter) {
+                is StatusFilter -> {
+                    statusFilter = filter.toUriPart() ?: "Tất cả"
+                }
+                is GenreFilter -> {
+                    genreFilter = filter.toUriPart()
+                }
+                else -> {}
+            }
+        }
+
+        url.addQueryParameter("status", statusFilter)
+        if (genreFilter.isNotEmpty()) {
+            url.addQueryParameter("tags", genreFilter)
+        }
+
+        val offset = (page - 1) * 40
+        url.addQueryParameter("offset", offset.toString())
+
+        return GET(url.toString(), headers)
+    }
+
+    override fun searchMangaSelector() = popularMangaSelector()
+
+    override fun searchMangaFromElement(element: Element) = popularMangaFromElement(element)
+
+    override fun mangaDetailsParse(document: Document) = SManga.create().apply {
+        title = document.selectFirst("h1.manga-title")!!.text()
+        description = document.selectFirst("div.manga-des")?.text()
+        status = document.selectFirst(".md-title:has(.fa-rss) ~ .md-content")?.text().toStatus()
+        genre = document.select("div.tags span a")?.joinToString { it.text() }
+        thumbnail_url = document.selectFirst("div.manga-img img")?.let { imageOrNull(it) }
+    }
+
+    override fun chapterListSelector() = ".manga .manga-chapters ul li:has(.chapter-name)"
+
+    override fun chapterFromElement(element: Element): SChapter {
+        return SChapter.create().apply {
+            element.select("div.chapter-name a").let {
+                name = it.text()
+                setUrlWithoutDomain(it.attr("abs:href"))
+            }
+            date_upload = parseChapterDate(element.select(":nth-child(3):last-child").text())
```

Comment:
You need the `>` prefix, otherwise it might catch another element in the structure. You should use `.selectFirst()`